jmix-flowui/flowui/src/main/java/io/jmix/flowui/sys/autowire/ReflectionCacheManager.java

```diff
@@ -0,0 +1,850 @@
+    }
+
+    protected Map<String, MethodHandle> getInstallTargetMethodsNotCached(Method[] uniqueDeclaredMethods) {
+        Map<String, MethodHandle> handlesMap = new HashMap<>();
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+        for (Method m : uniqueDeclaredMethods) {
+            if (Modifier.isPublic(m.getModifiers())
+                    && m.getParameterCount() == 1
+                    && (m.getName().startsWith("set") || m.getName().startsWith("add"))) {
+
+                Class<?> parameterType = m.getParameterTypes()[0];
+
+                if (Consumer.class.isAssignableFrom(parameterType)
+                        || Supplier.class.isAssignableFrom(parameterType)
+                        || Function.class.isAssignableFrom(parameterType)
+                        || parameterType.isInterface()) {
+
+                    m.trySetAccessible();
+
+                    MethodHandle methodHandle;
+                    try {
+                        methodHandle = lookup.unreflect(m);
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException("unable to get method handle " + m);
+                    }
+
+                    handlesMap.put(m.getName(), methodHandle);
+                }
+            }
+        }
+
+        return ImmutableMap.copyOf(handlesMap);
+    }
+
+    protected Map<SupplyMethodSignature, MethodHandle> getSupplyTargetMethodsNotCached(Method[] uniqueDeclaredMethods) {
+        Map<SupplyMethodSignature, MethodHandle> handlesMap = new HashMap<>();
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+
+        for (Method m : uniqueDeclaredMethods) {
+            if (Modifier.isPublic(m.getModifiers())
+                    && m.getParameterCount() == 1
+                    && m.getName().startsWith("set")
+                    && m.getName().contains("Renderer")) {
+
+                m.trySetAccessible();
+                MethodHandle methodHandle;
+                try {
+                    methodHandle = lookup.unreflect(m);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("unable to get method handle " + m);
+                }
+                SupplyMethodSignature supplyMethodSignature =
+                        new SupplyMethodSignature(m.getName(), Arrays.asList(m.getParameterTypes()));
+                handlesMap.put(supplyMethodSignature, methodHandle);
+            }
+        }
+
+        return ImmutableMap.copyOf(handlesMap);
+    }
+
+    @Nullable
+    protected Class<?> getAutowiringAnnotationClass(AnnotatedElement element) {
+        // TODO: kd, remove after major release
+        //  the @ViewComponent should be the only annotation for the fields that is processed by the framework.
+        //  currently, @Autowired is only needed for messageBundle bean. not removed for backward compatibility.
+        if (element.isAnnotationPresent(Autowired.class)) {
```

Comment:
`@Autowired` annotation was used only for `messageBundle` injection. It was decided to leave only one annotation.

This code is kept for backward compatibility.
Must be removed after major release.

I am attaching a link to the issue in the studio: https://youtrack.jmix.io/issue/JST-4969/Replace-injection-annotation-for-messageBundle